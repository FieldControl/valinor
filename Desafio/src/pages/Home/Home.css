.container {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 100vw;
}

.container h1 {
  margin: 40px 0 24px;
  color: #fff;
}

.noResearch {
  margin-top: 20px;
  color: #fff;
  font-size: 25px;
  font-weight: 700;
}

.results {
  margin-top: 0.5rem;
  color: #fff;
  font-size: 15px;
  font-weight: 700;
  background: #1c1c1c;
  border-radius: 5px;
  padding: 7px;
}
