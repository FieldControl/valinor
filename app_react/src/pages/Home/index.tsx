import Header from "../../components/Header";
import RepoCard from "../../components/RepoCard";
import Pagination from "../../components/Pagination";
import { useSearch } from "../../hooks/useSearchRepo";
import {
  PaginationData,
  paginateData,
  calculateTotalPages,
  reposPerPage,
} from "../../utils/paginationUtils";

export default function Home() {
  const {
    repos,
    currentPage: searchCurrentPage,
    setCurrentPage,
    loading,
  } = useSearch();

  const { currentPage, data: currentRepos }: PaginationData<(typeof repos)[0]> =
    paginateData(repos, searchCurrentPage, reposPerPage);
  const totalPages = calculateTotalPages(repos.length, reposPerPage);
  return (
    <>
      <Header />
      <div className="p-4 flex align-middle">
        <main className="w-full">
          {loading ? (
            // Spinner do Tailwind enquanto os dados estão sendo carregados
            <div className="flex items-center justify-center h-screen">
              <div className="animate-spin rounded-full h-16 w-16 border-b-2 border-gray-300"></div>
            </div>
          ) : (
            // Renderizar os dados
            <>
              {currentRepos.length === 0 ? (
                <p className="text-white text-3xl h-[100vh] items-center flex justify-center">
                  Nenhum repositório encontrado.
                </p>
              ) : (
                currentRepos.map((item) => (
                  <RepoCard
                    key={item.full_name}
                    owner={item.owner}
                    full_name={item.full_name}
                    language={item.language}
                    updated_at={item.updated_at}
                    watchers_count={item.watchers_count}
                    description={item.description}
                    topics={item.topics}
                    html_url={item.html_url}
                  />
                ))
              )}
              {currentRepos.length > 0 && (
                <Pagination
                  currentPage={currentPage}
                  totalPages={totalPages}
                  setCurrentPage={setCurrentPage}
                />
              )}
            </>
          )}
        </main>
      </div>
    </>
  );
}
