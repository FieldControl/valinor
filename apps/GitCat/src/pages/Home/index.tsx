import React, { useState } from 'react'
import InputSearch from '../../components/InputSearch'
import { useQuery } from '../../contexts/queryContext'

import * as S from './styles'

function Home () {

    const [searchQuery, setSearchQuery] = useState('')
    const {setRequestQuerys} = useQuery()

    function sendQuery () {
        setRequestQuerys({page: 1, search: searchQuery, sort: 'full_name', typeSearch: 'repositories' })
    }

    return (
        <S.Container>
            <S.Main>
                <S.Logo />
                <InputSearch
                    value={searchQuery}
                    change={setSearchQuery}
                    submit={sendQuery}
                    placeholder="Search a repository"
                />
            </S.Main>
        </S.Container>
    )
}

export default Home