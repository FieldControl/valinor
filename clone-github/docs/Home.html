<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>JSDoc: Class: Home</title>

    <script src="scripts/prettify/prettify.js"></script>
    <script src="scripts/prettify/lang-css.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link
      type="text/css"
      rel="stylesheet"
      href="styles/prettify-tomorrow.css"
    />
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css" />
  </head>

  <body>
    <div id="main">
      <h1 class="page-title">Class: Home</h1>

      <section>
        <header>
          <h2>
            <span class="attribs"><span class="type-signature"></span></span
            >Home<span class="signature">()</span
            ><span class="type-signature"></span>
          </h2>

          <div class="class-description"><p>Container Home.</p></div>
        </header>

        <article>
          <div class="container-overview">
            <h2>Constructor</h2>

            <h4 class="name" id="Home">
              <span class="type-signature"></span>new Home<span
                class="signature"
                >()</span
              ><span class="type-signature"></span>
            </h4>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="containers_Home_Home.js.html"
                      >containers/Home/Home.js</a
                    >, <a href="containers_Home_Home.js.html#line24">line 24</a>
                  </li>
                </ul>
              </dd>
            </dl>

            <h5>Example</h5>

            <pre class="prettyprint"><code>&lt;Home></code></pre>
          </div>
        </article>
      </section>
    </div>

    <nav>
      <h2><a href="index.html">Home</a></h2>
      <h3>Classes</h3>
      <ul>
        <li><a href="Home.html">Home</a></li>
      </ul>
      <h3>Global</h3>
      <ul>
        <li><a href="global.html#fetchRepos">fetchRepos</a></li>
        <li><a href="global.html#formatNumber">formatNumber</a></li>
        <li>
          <a href="global.html#formatNumberWithComma">formatNumberWithComma</a>
        </li>
        <li><a href="global.html#formatString">formatString</a></li>
        <li><a href="global.html#formatUrl">formatUrl</a></li>
        <li><a href="global.html#getLinksOfHeader">getLinksOfHeader</a></li>
        <li><a href="global.html#IssueCard">IssueCard</a></li>
        <li><a href="global.html#Navbar">Navbar</a></li>
        <li><a href="global.html#Paginator">Paginator</a></li>
        <li><a href="global.html#regexLink">regexLink</a></li>
        <li><a href="global.html#RepoCard">RepoCard</a></li>
        <li><a href="global.html#Search">Search</a></li>
      </ul>
    </nav>

    <br class="clear" />

    <footer>
      Documentation generated by
      <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.6</a> on Tue May 11
      2021 17:37:09 GMT-0300 (Brasilia Standard Time)
    </footer>

    <script>
      prettyPrint();
    </script>
    <script src="scripts/linenumber.js"></script>
  </body>
</html>
