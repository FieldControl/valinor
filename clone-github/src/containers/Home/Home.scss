.home {
  display: flex;
  flex-direction: column;
}
.home__container {
  box-sizing: border-box;
  width: 60%;
  margin: 20px auto;
  padding: 20px;
  & .home__total__count {
    text-align: left;
    padding: 20px;
  }
}
.home__search {
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgb(56, 56, 56);
  height: 50px;
}
.home__button {
  height: 28px;
  box-sizing: border-box;
  padding: 1px;
  border-color: transparent;
  box-shadow: 0 2px 3px silver;
  border-top: 1px solid gray;
  background-color: rgb(56, 56, 56);
  color: white;
  margin-left: 10px;
  border-radius: 5px;
  padding: 5px;
  font-family: AssistantRegular, serif;
  letter-spacing: 1px;
  &:hover {
    cursor: pointer;
    opacity: 0.7;
  }
}
.home__title {
  box-sizing: border-box;
  text-align: center;
  margin: 50px 0;
  font-weight: bold;
  font-size: 2.2rem;
  font-family: AssistantRegular, serif;
  letter-spacing: 1px;
}
.home__total__count {
  letter-spacing: 1px;
}

@media (max-width: 820px) {
  .home__container {
    width: 100%;
  }
  .home__search {
    width: 100%;
  }
  .home__title {
    font-size: 1.2rem;
  }
}
