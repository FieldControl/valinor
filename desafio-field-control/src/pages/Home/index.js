import { useEffect, useState } from "react";
import {
  Container,
  Titulo,
  DivInput,
  Input,
  Button,
  ListaRepo,
  ContainerLinguagem,
  ContainerDescricao,
  ContainerLista,
  RepositorioImg,
  Informacoes,
  ParagrafosDesc,
  TopicosLinha,
  Topicos,
  LinhaInfos,
  ParagrafosInfo,
  ContadorPaginacao,
} from "./style";
import { GoRepo } from "react-icons/go";
import { FaRegStar, FaRegCircle } from "react-icons/fa";

const Home = () => {
  const [paginacaoRepositorio, setPaginacaoRepositorio] = useState([]);
  const [linguagem, setLinguagem] = useState("node");
  const [linguagemTemp, setLinguagemTemp] = useState("");
  const [pagina, setPagina] = useState(1);
  const [sobreLinguagem, setSobreLinguagem] = useState([]);

  const generateID = () => Math.round(Math.random() * 1000);
  console.log(generateID());

  useEffect(() => {
    fetch(
      `https://api.github.com/search/repositories?q=${linguagem}&page=${pagina}&per_page=10`
    )
      .then((response) => response.json())
      .then((data) => {
        setPaginacaoRepositorio(data.items);
      });
  }, [linguagem, pagina]);

  useEffect(() => {
    fetch(`https://api.github.com/search/topics?q=${linguagem}&per_page=1`)
      .then((response) => response.json())
      .then((data) => {
        setSobreLinguagem(data.items);
      });
  }, [linguagem]);

  const atualizaPesquisa = () => {
    setPagina(1);
    setLinguagem(linguagemTemp);
    setLinguagemTemp("");
  };

  const retornoLinguagem = (linguagem) => {
    if (linguagem === null) {
      return <></>;
    } else {
      return (
        <div>
          <FaRegCircle />
          <p>{linguagem}</p>
        </div>
      );
    }
  };

  const paginaAnterior = () => {
    if (pagina > 1) {
      setPagina(pagina - 1);
    }
  };

  const proximaPagina = () => {
    if (pagina >= 1 && paginacaoRepositorio.length === 10) {
      setPagina(pagina + 1);
    }
  };

  return (
    <Container>
      <Titulo>Repositórios</Titulo>
      <DivInput>
        <Input
          type="text"
          onChange={(e) => setLinguagemTemp(e.target.value)}
          placeholder="Digite a tecnologia"
          value={linguagemTemp}
          data-testid="input"
        ></Input>
        <Button data-testid="botao-pesquisa" onClick={atualizaPesquisa}>
          Pesquisar
        </Button>
      </DivInput>
      <ListaRepo>
        <ContainerLinguagem data-testid="linguagem">
          {sobreLinguagem.map((sobre) => {
            return (
              <ContainerDescricao key={generateID}>
                <Titulo>{sobre.name}</Titulo>
                <p>{sobre.short_description}</p>
              </ContainerDescricao>
            );
          })}
        </ContainerLinguagem>
        {paginacaoRepositorio.map((repo) => {
          return (
            <ContainerLista key={repo.id}>
              <RepositorioImg>
                <GoRepo />
              </RepositorioImg>

              <Informacoes>
                <a href={repo.url}>
                  <h2>{repo.full_name}</h2>
                </a>
                <ParagrafosDesc>{repo.description}</ParagrafosDesc>

                <TopicosLinha>
                  {repo.topics.map((rep) => {
                    return (
                      <Topicos>
                        <a href={`https://github.com/topics/${rep}`}>
                          <p>{rep}</p>
                        </a>
                      </Topicos>
                    );
                  })}
                </TopicosLinha>
                <LinhaInfos>
                  <ParagrafosInfo>
                    <div>
                      <FaRegStar />
                      <p>{repo.watchers}</p>
                    </div>
                    {retornoLinguagem(repo.language)}
                  </ParagrafosInfo>
                </LinhaInfos>
              </Informacoes>
            </ContainerLista>
          );
        })}
      </ListaRepo>
      <ContadorPaginacao>
        <Button onClick={paginaAnterior} data-testid="pagina-anterior">
          Anterior
        </Button>

        <Button onClick={proximaPagina} data-testid="pagina-seguinte">
          Próxima
        </Button>
      </ContadorPaginacao>
    </Container>
  );
};

export default Home;
