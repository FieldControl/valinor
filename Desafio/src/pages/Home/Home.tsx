import { useEffect, useState } from "react";
import Cards from "../../components/Cards/Cards";
import Search from "../../components/Search/Search";
import Pagination from "../../components/Pagination/Pagination";
import "./Home.css";

const limit = 12;

function Home() {
  const [repositories, setRepositories] = useState([]);
  const [search, setSearch] = useState([]);
  const [totalCount, setTotalCount] = useState([]);
  const [offset, setOffset] = useState(0);

  function handleSearch(data) {
    setSearch(data.input);
  }

  useEffect(() => {
    const query = {
      page: {
        limit,
        offset,
      },
    };

    if (search) {
      query.filter = {
        search,
      };
    }

    console.log(offset);

    fetch(
      `https://api.github.com/search/repositories?q=${search}&per_page=12&page=${offset}`
    )
      .then((response) => response.json())
      .then((data) => {
        setTotalCount(data.total_count);
        setRepositories(data.items);
      })
      .catch((error) => console.log(error));
  }, [search, offset]);

  return (
    <div className="container">
      <h1>Pesquisa GitHub</h1>
      <Search onSearch={handleSearch} />
      {search.length === 0 && (
        <div className="noResearch">Sem repositórios, faça sua pesquisa...</div>
      )}
      {totalCount >= 1 && (
        <div className="results">Total de resultado: {totalCount}</div>
      )}
      <Cards data={repositories} />
      {totalCount && (
        <Pagination limit={limit} offset={offset} setOffset={setOffset} />
      )}
    </div>
  );
}

export default Home;
