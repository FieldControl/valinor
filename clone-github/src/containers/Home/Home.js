import React, { Component } from "react";
import "./Home.scss";
import Search from "../../components/Search/Search";
import { withRouter } from "react-router-dom";
import RepoCard from "./RepoCard/RepoCard";
import IssueCard from "./IssueCard/IssueCard";

import { formatNumberWithComma } from "../../helpers/format";

class Home extends Component {
  state = {
    repos: [],
    search: "",
    submit: false,
    pageCounter: 0,
    totalCount: 0,
    type: "repositories",
    issues: [],
  };
  /* eslint-disable  react/prop-types */

  componentDidUpdate(prevProps, prevState) {
    const { submit } = this.state;
    console.log(prevProps);
    console.log(prevState);

    if (submit) {
      this.fetchRepos();
    }
  }
  fetchRepos = async () => {
    const { search, type } = this.state;
    try {
      const response = await fetch(
        `https://api.github.com/search/${type}?q=${search}&page=1&per_page=10`
      );
      const data = await response.json();
      if (type === "issues") {
        this.setState({
          issues: data.items,
          submit: false,
          totalCount: data.total_count,
        });
      }
      if (type === "repositories") {
        this.setState({
          repos: data.items,
          submit: false,
          totalCount: data.total_count,
        });
      }
    } catch (err) {
      console.log(err);
      this.setState({
        repos: [],
        issues: [],
        submit: false,
      });
    }
  };
  searchField = (e) => {
    this.setState({ search: e.target.value });
  };
  searchType = (e) => {
    this.setState({ type: e.target.value });
  };
  submitSearch = () => {
    const { search, type } = this.state;
    if (search.length && type.length) {
      this.setState({ submit: true });
    }
  };

  render() {
    const { search, type, issues, repos, totalCount } = this.state;
    let name = "";
    if (repos.length || issues.length) {
      name = (
        <ul className="home__container">
          <h3 className="home__total__count">
            {" "}
            {formatNumberWithComma(totalCount)} {`${type} results`}
          </h3>

          {type === "repositories"
            ? repos.map((repo, index) => {
                return <RepoCard key={index} {...repo} />;
              })
            : null}

          {type === "issues"
            ? issues.map((issue, index) => {
                return <IssueCard key={index} {...issue} />;
              })
            : null}
        </ul>
      );
    }
    return (
      <>
        <div className="home">
          <div className="home__search">
            <Search
              change={(e) => this.searchField(e)}
              text={search}
              changeType={(e) => this.searchType(e)}
              type={type}
              click={() => this.submitSearch()}
            />
            <button
              className="home__button"
              onClick={() => this.submitSearch()}
            >
              {" "}
              Search
            </button>
          </div>

          {name}
        </div>
      </>
    );
  }
}

export default withRouter(Home);
