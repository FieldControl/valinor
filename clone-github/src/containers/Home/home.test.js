import { render, cleanup } from "@testing-library/react";
import Home from "./Home.js";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

afterEach(cleanup);

test("should show Search ", () => {
  const history = createMemoryHistory();
  history.push("/");
  render(
    <Router history={history}>
      <Home />
    </Router>
  );
  // expect(screen.getByText(/Search/i)).toBeInTheDocument();
  expect(<Home />).toMatchSnapshot();
});
