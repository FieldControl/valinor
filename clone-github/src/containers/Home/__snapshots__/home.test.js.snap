// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show Search  1`] = `<withRouter(Home) />`;
